### PR TITLE
Add unicode completions to markdown files.

### PR DESCRIPTION
**Pull request by EdAyers**
This is my first ever PR so be gentle.

Changed behaviour:

- A new setting `lean.input.markdown`. When it is true, the lean abbreviation provider (in `input.ts`) will also work in markdown (.md) files.
- Add `<>` replacement. This is because markdown auto matches `<` so you get weird behaviour otherwise.

### CODE REVIEW

**Comment by EdAyers** [comment]
Also fixes #89

File: src/extension.ts [hunk 1]
```diff
@@ -62,10 +62,17 @@ export function activate(context: ExtensionContext) {
 
     // Register support for unicode input.
     (async () => {
-        const translations: any = await loadJsonFile.default(context.asAbsolutePath('translations.json'));
+        const translations : any = await loadJsonFile.default(context.asAbsolutePath('translations.json'));
+        const modes = [LEAN_MODE];
+        const input_markdown = workspace.getConfiguration('lean.input').get('markdown');
```
Comment (gebner) [inline src/extension.ts]: Please install the tslint extension in vscode, and fix all errors.

File: src/extension.ts [hunk 2]
```diff
@@ -62,10 +62,17 @@ export function activate(context: ExtensionContext) {
...
+        if (input_markdown) {modes.push(MARKDOWN_MODE);}
```
Comment (gebner) [inline src/extension.ts]: Please move the code that reads the desired modes from the configuration to `input.ts`.  Also make sure that the modes are reread on configuration change events (there are already event listeners installed).

Look at `inputModeLeader()` for inspiration.

File: package.json [hunk 3]
```diff
@@ -60,6 +60,11 @@
 					"default": true,
 					"description": "Enable Lean input mode"
 				},
+				"lean.input.markdown" : {
```
Comment (gebner) [inline package.json]: I'd prefer a `lean.input.languages` configuration option that takes a list as argument, defaulting to `["lean"]`.

File: src/extension.ts (hunk 1 above)
Comment (gebner) [reply src/extension.ts]: You can also do `npm run lint` to see all linting errors at once.

File: src/extension.ts [hunk 4]
```diff
@@ -62,10 +62,17 @@ export function activate(context: ExtensionContext) {
...
+        const hover_providers = modes.map(mode =>   
+            languages.registerHoverProvider(mode, new LeanInputExplanationHover(translations))
+        );
+
         context.subscriptions.push(
-            languages.registerHoverProvider(LEAN_MODE, new LeanInputExplanationHover(translations)),
-            new LeanInputAbbreviator(translations, LEAN_MODE));
+            ...hover_providers,
```
Comment (gebner) [inline src/extension.ts]: I hope that we can get by with a single `LeanInputExplanationHover` if we register it for `{}: DocumentFilter`.

Comment (EdAyers) [reply src/extension.ts]: I couldn't get this to work :(

Comment (EdAyers) [reply src/extension.ts]: I'm now doing `languages.registerHoverProvider(inputLanguages, new LeanInputExplanationHover(translations));` in `extension.ts` where `inputLanguages` is retieved from settings and is eg `["lean", "markdown"]`. So you have to reload if you change `lean.input.languages` and want to get hover hints for the new language.

Comment (EdAyers) [reply src/extension.ts]: Only the hovers though

File: package.json (hunk 3 above)
Comment (EdAyers) [reply package.json]: done

File: src/extension.ts (hunk 1 above)
Comment (EdAyers) [reply src/extension.ts]: done

File: src/extension.ts (hunk 2 above)
Comment (EdAyers) [reply src/extension.ts]: done

File: package.json [hunk 5]
```diff
@@ -60,10 +60,19 @@
...
+					"type": "array",
+					"default": ["lean"],
+					"description": "Enable Lean unicode input in other file types."
+				},
 				"lean.input.leader": {
 					"type": "string",
 					"default": "\\",
-					"description": "Leader key to trigger input mode"
+					"description": "Leader key to trigger input mode",
+					"items": {
```
Comment (gebner) [inline package.json]: The `items` description is in the wrong configuration setting, isn't it?

File: src/extension.ts [hunk 6]
```diff
@@ -62,10 +62,13 @@ export function activate(context: ExtensionContext) {
 
     // Register support for unicode input.
     (async () => {
-        const translations: any = await loadJsonFile.default(context.asAbsolutePath('translations.json'));
+        const translations: any = await loadJsonFile(context.asAbsolutePath('translations.json'));
+        const inputLanguages: string[] = workspace.getConfiguration('lean.input').get('languages', ['lean']);
```
Comment (gebner) [inline src/extension.ts]: Please use the `inputModeLanguages()` function you added to `input.ts`.  (You need to add the `export` modifier to use it from other files.)

File: src/extension.ts (hunk 4 above)
Comment (gebner) [reply src/extension.ts]: Ok, it was worth a try.

**Comment by EdAyers** [comment]
Oops I accidentally closed it.

**Comment by EdAyers** [comment]
This is awesome.

**Comment by gebner** [comment]
And deployed: 01944cbe55a93540554f8346c0aa0982222c87c0

**Comment by EdAyers** [comment]
@gebner I've also got a branch that lets you specify your own custom translations. Would you be interested in me PRing that?

**Comment by gebner** [comment]
@EdAyers Sure.

**Pull request merged**